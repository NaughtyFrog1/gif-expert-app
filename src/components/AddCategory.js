import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    const trimInput = inputValue.trim()

    if (trimInput.length < 2) return

    setCategories((categories) => [...categories, trimInput])
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        placeholder="Add category"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      ></input>
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
