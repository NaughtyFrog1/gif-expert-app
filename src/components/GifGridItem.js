import React from 'react'

const GifGridItem = ({id, title, url}) => {
  return (
    <li className="card">
      <img src={url} alt={title} />
    </li>
  )
}

export default GifGridItem
