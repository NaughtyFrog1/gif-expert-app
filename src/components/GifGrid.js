import React from 'react'
import PropTypes from 'prop-types'
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category)

  return (
    <div className="category">
      <h3>{category}</h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}

      <ol className="gif-cards">
        {gifs.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </ol>
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifGrid
