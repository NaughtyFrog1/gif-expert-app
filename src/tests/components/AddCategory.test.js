// eslint-disable-next-line no-unused-vars
import { shallow, ShallowWrapper } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('test for AddCategory', () => {
  const setCategories = jest.fn()

  /** @type {ShallowWrapper} */
  let wrapper

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('should render the component properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should change the input value', () => {
    const value = 'New value'
    wrapper.find('input').simulate('change', { target: { value } })

    expect(wrapper.find('input').prop('value')).toBe('New value')
  })

  test("shouldn't post the information with the submit", () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} })
    expect(setCategories).not.toHaveBeenCalled()
  })

  test('should call setCategories and clear the text box', () => {
    wrapper.find('input').simulate('change', { target: { value: 'New value' } })
    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).toHaveBeenCalled()
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})
