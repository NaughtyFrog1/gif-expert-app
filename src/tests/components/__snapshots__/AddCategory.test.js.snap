// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for AddCategory should render the component properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add category"
    type="text"
    value=""
  />
</form>
`;
